src/DotPulsar/Internal/ConsumerChannel.cs

```diff
@@ -155,6 +174,7 @@ public async ValueTask DisposeAsync()
         {
             try
             {
+                _cancellationTokenSource.Dispose();
```

Comment:
Umm, did you mean .Cancel()?  I don't see any linked token sources or registrations that would require disposal, and disposal doesn't seem to actually cancel the tokens either, so your loop staying running forever.

Reply:
good catch! I usually add Dispose anyway, as I got bitten several times by not adding it and then later adding functionality which requires it...